src/ContentRepository/Fields/AllRolesField.cs

```diff
@@ -20,9 +23,21 @@ protected override void ImportData(XmlNode fieldNode, ImportContext context)
 
         public override object GetData()
         {
-            // this method will load only containers that the current user has permissions for
-            return Node.LoadNodes(Providers.Instance.SecurityHandler.SecurityContext
+            var thisPath = this.Content.Path;
+            var orgUnitNt = Providers.Instance.StorageSchema.NodeTypes[nameof(OrganizationalUnit)];
+
+            // loads only containers that the current user has permissions for
+            var allParents = Node.LoadNodes(Providers.Instance.SecurityHandler.SecurityContext
                 .GetParentGroups(this.Content.Id, false)).ToArray();
+
+            // valid item is all ancestors or any node that is not an OrganizationalUnit
+            var filtered = allParents
+                .Where(n => !n.NodeType.IsInstaceOfOrDerivedFrom(orgUnitNt) ||
+                            thisPath.StartsWith(n.Path, StringComparison.OrdinalIgnoreCase))
```

Comment:
There is a tool method for this:
```suggestion
                            RepositoryPath.IsInTree(n.Path, thisPath))
```